backend/models.py

```diff
@@ -0,0 +1,163 @@
+import enum
+from datetime import datetime
+from typing import List, Union, Optional
+
+from sqlalchemy import (
+    select, Enum, DateTime,
+    Boolean, Column, ForeignKey, 
+    Integer, String, Float)
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm.session import Session
+from sqlalchemy.ext.declarative import declared_attr
+
+from .database import Base
+
+class Value(Base):
+    __abstract__ = True
+    
+    id = Column(Integer, primary_key=True, index=True)
+    @declared_attr
+    def entity_id(cls):
+        return Column(Integer, ForeignKey('entities.id'))
+
+    @declared_attr
+    def attribute_id(cls):
+        return Column(Integer, ForeignKey('attributes.id'))
+
+
+class ValueBool(Value):
+    __tablename__ = 'values_bool'
+    value = Column(Boolean)
+
+
+class ValueInt(Value):
+    __tablename__ = 'values_int'
+    value = Column(Integer)
+
+
+class ValueFloat(Value):
+    __tablename__ = 'values_float'
+    value = Column(Float)
+
+
+class ValueForeignKey(Value):
+    __tablename__ = 'values_fk'
+    value = Column(Integer)
+
+
+class ValueStr(Value):
+    __tablename__ = 'values_str'
+    value = Column(String)
+
+
+class ValueDatetime(Value):
+    __tablename__ = 'values_datetime'
+    value = Column(DateTime)
+
+
+class AttrType(enum.Enum):
```

Comment:
It makes sense, I was also thinking about having everything in one place. Actually, I dropped in your definition of `AttrType` and none of my tests broke 😏. It will require to rewrite some stuff in crud.py, but I think we can go with approach like you suggested